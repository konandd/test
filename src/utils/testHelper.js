import renderer from 'react-test-renderer';

const { expect } = global;
export const toMatchSnapshot = (name, component) => {
  expect(
    renderer
      .create(component)
      .toJSON(),
  ).toMatchSnapshot('react-test-renderer');
};
