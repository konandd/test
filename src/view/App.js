import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { showMessage } from '../redux/message/actions';
import { decrease, increase } from '../redux/counter/actions';
import { getTheme, switchTheme } from '../redux/theme/actions';
import { getFirms, getLaptops } from '../redux/products/actions';
import Navigation from '../components/Navigation';
import ProductList from '../components/ProductList';
import './style.css';


class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
    };
    this.initRedux();
  }


    initRedux = () => {
      const { actions: { getTheme, getFirms, getLaptops } } = this.props;
      getTheme();
      getFirms();
      getLaptops();
    };

    render() {
      const {
        value, theme, firms, laptops,
        actions: {
          showMessage, increase, decrease, switchTheme,
        },
      } = this.props;

      const themeCol = {
        darck: {
          background: '#4d5459',
        },
        light: {
          background: '#ffffff',
        },
      };

      return (
        <React.Fragment>
          <div style={theme ? themeCol.darck : themeCol.light}>
            <h1 className="app_root">Test</h1>
            <p>{value}</p>
            <button type="button" onClick={() => showMessage({ text: 'fwef' })}>popka</button>
            <button type="button" onClick={() => increase(value)}>increase</button>
            <button type="button" onClick={() => decrease(value)}>decrease</button>
            <button type="button" onClick={() => switchTheme(theme)}>theme</button>
          </div>
          <Navigation firms={firms} laptops={laptops}>
            <ProductList />
          </Navigation>
        </React.Fragment>
      );
    }
}

function mapStateToProps(state) {
  return {
    value: state.counter.value,
    theme: state.theme.theme,
    firms: state.products.firms,
    laptops: state.products.laptops,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators(
      {
        showMessage,
        increase,
        decrease,
        switchTheme,
        getTheme,
        getFirms,
        getLaptops,
      },
      dispatch,
    ),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(App);
