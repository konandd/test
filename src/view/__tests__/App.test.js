import React from 'react';
import {toMatchSnapshot} from "../../utils/testHelper";
import App from "../App";

it('Test run', () =>
    toMatchSnapshot('Test run for App', <App/>)
);
