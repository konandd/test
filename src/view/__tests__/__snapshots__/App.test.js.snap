// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react-test-renderer 1`] = `
<h1
  className="app_root"
>
  Test
</h1>
`;
